pandapower/plotting/plotting_toolbox.py

```diff
@@ -164,6 +165,35 @@ def coords_from_node_geodata(element_indices, from_nodes, to_nodes, node_geodata
     return coords, elements_with_geo
 
 
+def set_line_geodata_from_bus_geodata(net, line_index=None, overwrite=False):
+    """
+    Sets coordinates in net.line_geodata based on the from_bus and to_bus x,y coordinates
+    in net.bus_geodata
+    :param net: pandapowerNet
+    :param line_index: index of lines, coordinates of which will be set from bus geodata (all lines if None)
+    :param overwrite: whether the existing coordinates in net.line_geodata must be overwritten
+    :return: None
+    """
+    line_index = line_index if line_index is not None else net.line.index
+    if not overwrite:
+        line_index = np.setdiff1d(line_index, net.line_geodata.index)
+
+    coords, line_index_successful = coords_from_node_geodata(element_indices=line_index,
+                                                             from_nodes=net.line.loc[line_index, 'from_bus'].values,
+                                                             to_nodes=net.line.loc[line_index, 'to_bus'].values,
+                                                             node_geodata=net.bus_geodata,
+                                                             table_name="line_geodata", node_name="bus_geodata")
+
+    net.line_geodata.drop(np.intersect1d(net.line_geodata.index, line_index_successful), inplace=True)
+    net.line_geodata = pd.concat([net.line_geodata, pd.DataFrame(index=line_index_successful,
+                                               columns=["coords"], data=np.nan)]).reindex(index=net.line.index)
```

Comment:
`reindex` will fill all missing values with `np.NaN`. Since `index_line_successful` should be a subset of `net.line.index`, `net.line_geodata.reindex(net.line.index)` would be sufficient, wouldn't it?

Reply:
thank you, now the code is easier to read